src/main/java/frc/robot/subsystems/drive/DriveIOCTRE.java

```diff
@@ -54,6 +87,42 @@ public void updateInputs(DriveIOInputs inputs) {
     inputs.gyroRate = super.getPigeon2().getAngularVelocityZWorld().getValue();
     inputs.operatorForwardDirection = super.getOperatorForwardDirection();
     inputs.odometryIsValid = super.isOdometryValid();
+
+    this.odometryLock.lock();
+
+    inputs.timestamp = this.timestampQueue.stream().mapToDouble(Double::valueOf).toArray();
+    this.timestampQueue.clear();
+
+    inputs.gyroYaw = this.gyroYawQueue.stream().toArray(Rotation2d[]::new);
+    this.gyroYawQueue.clear();
+
+    for (int i = 0; i < this.getModules().length; i++) {
+      inputs.drivePositions[i] =
+          this.drivePositionQueues.get(i).stream().mapToDouble(Double::valueOf).toArray();
+      inputs.steerPositions[i] =
+          this.steerPositionQueues.get(i).stream().toArray(Rotation2d[]::new);
+
+      this.drivePositionQueues.get(i).clear();
+      this.steerPositionQueues.get(i).clear();
+    }
+
+    this.odometryLock.unlock();
+  }
+
+  private void updateTelemetry(SwerveDriveState state) {
```

Comment:
add a comment to this method and the one above explaining what is going on. these are critical methods

Reply:
Needs a check